### PR TITLE
Preparing v0.6.0

### PR DESCRIPTION
**Pull request by oualib**
- class AutoDataPrep, AutoClustering, AutoML -> delphi
 - vmodel -> contour, to_python, to_sql
 - vcolumn: cut
 - new datasets: gapminder, pop_growth
 - vdataframe: ncols_block for agg, describe ... function animated + contour
 - tqdm -> installation
 - fonctions: gen_dataset, gen_regular_dataset
 - bayesian_search, enet_search_cv, randomized_features_search_cv, stepwise

### CODE REVIEW

File: verticapy/datasets.py [hunk 1]
```diff
@@ -49,13 +49,168 @@
...
+                       which are representing the lower and the upper bound
+        For date     : The subdictionary must include two keys. 'type' must
+                       be set to 'date' and 'range' must include the start
+                       date and the number of days after.
+        For datetime : The subdictionary must include two keys. 'type' must
+                       be set to 'date' and 'range' must include the start
+                       date and the number of days after.
+cursor: DBcursor, optional
+    Vertica DB cursor.
+
```
Comment (afard) [inline verticapy/datasets.py:88]: Description for `nrows`?

File: verticapy/learn/delphi.py [hunk 2]
```diff
@@ -0,0 +1,835 @@
...
+    def __init__(self,
+                 name: str,
+                 cursor=None,
+                 estimator: (list, str) = "fast",
+                 estimator_type: str = "auto",
+                 metric: str = "auto",
+                 cv: int = 3,
+                 pos_label: (int, float, str) = None,
+                 cutoff: float = -1,
+                 nbins: int = 1000,
```
Comment (afard) [inline verticapy/learn/delphi.py:588]: The default value of `nbins` seems to be very big. I wonder what has been your criteria for picking the default values.

File: verticapy/learn/delphi.py [hunk 3]
```diff
@@ -0,0 +1,835 @@
...
+                    self.parameters["estimator"] = [LinearSVR(self.name, cursor=self.cursor), ElasticNet(self.name, cursor=self.cursor), LinearRegression(self.name, cursor=self.cursor), Lasso(self.name, cursor=self.cursor), Ridge(self.name, cursor=self.cursor), RandomForestRegressor(self.name, cursor=self.cursor), XGBoostRegressor(self.name, cursor=self.cursor),]
+                elif self.parameters["estimator"] == "fast":
+                    self.parameters["estimator"] = [LinearRegression(self.name, cursor=self.cursor), ElasticNet(self.name, cursor=self.cursor), RandomForestRegressor(self.name, cursor=self.cursor),]
+                else:
+                    self.parameters["estimator"] = [LinearSVR(self.name, cursor=self.cursor), ElasticNet(self.name, cursor=self.cursor), LinearRegression(self.name, cursor=self.cursor), Lasso(self.name, cursor=self.cursor), Ridge(self.name, cursor=self.cursor), RandomForestRegressor(self.name, cursor=self.cursor), XGBoostRegressor(self.name, cursor=self.cursor), KNeighborsRegressor(self.name, cursor=self.cursor),]
+            elif self.parameters["estimator_type"].lower() in ("multi", "auto",):
```
Comment (afard) [inline verticapy/learn/delphi.py:694]: It seems that "auto" is always treated the same as "multi"! Then, how have you achieved auto detection?

File: verticapy/learn/delphi.py [hunk 4]
```diff
@@ -0,0 +1,835 @@
...
+                self.parameters["estimator_type"] = "binary"
```
Comment (afard) [inline verticapy/learn/delphi.py:695]: Why have you changed `self.parameters["estimator_type"]` to `"binary"`?

File: verticapy/learn/delphi.py (hunk 2 above)
Comment (oualib) [reply verticapy/learn/delphi.py:588]: Done; I put it to 100, I changed it for testing purpose.

File: verticapy/datasets.py (hunk 1 above)
Comment (oualib) [reply verticapy/datasets.py:88]: I added the description.

File: verticapy/learn/delphi.py (hunk 3 above)
Comment (oualib) [reply verticapy/learn/delphi.py:694]: It's an 'elif', you should look at the previous one.

File: verticapy/learn/delphi.py (hunk 4 above)
Comment (oualib) [reply verticapy/learn/delphi.py:695]: Mistake for multi which is now corrected.

File: verticapy/learn/vmodel.py [hunk 5]
```diff
@@ -2292,6 +2363,527 @@ def to_sklearn(self):
...
+            func += "singular_values = np.array({})\n".format(value)
+            func += "\tright_singular_vectors = np.array({})\n".format(sv)
+            func += "\tL = []\n"
+            n = len(sv[0])
+            func += "\tfor i in range({}):\n".format(n)
+            func += "\t\tL += [np.sum(X * right_singular_vectors[i] / singular_values[i], axis=1)]\n"
+            func += "\tresult = np.column_stack(L)\n"
+            func += "\treturn result\n"
+            return func
+            """
```
Comment (afard) [inline verticapy/learn/vmodel.py:2484]: Why is this block of code commented out?

File: verticapy/tests/vModel/test_delphi.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    winequality = load_winequality(cursor=base.cursor)
+    yield winequality
+    with warnings.catch_warnings(record=True) as w:
+        drop(
+            name="public.winequality", cursor=base.cursor,
+        )
+
+class TestDelphi:
+
+    def test_AutoML(self, base, titanic_vd):
```
Comment (afard) [inline verticapy/tests/vModel/test_delphi.py:60]: The test of AutoML does not cover all of its different parameters.

**Review by afard — APPROVED** [review]

**Pull request merged**